### ISSUE TITLE
When upgrading in rolling upgrade mode (1.25.7 to 1.26.9), some pods are not restarting due to authentication failure 

### ISSUE THREAD

**Issue by lallema** [kubernetes/kubernetes#122390]
### What happened?

I have a cluster in 1.25.7 version and upgrade it to 1.26.9.
When all is finished, some pods are not able to get the images in our private registry (docker) as if the pod lost the secret.
Knowing that other pods are correctly restarted in the same namespace (and secret needed to connect to the private registry are present in this namespace)

in my case:
knative-operator-6649bc5fc7-jlk8q                             0/1     ImagePullBackOff   0             14m
operator-webhook-5fd954994f-wp9kv                             0/1     ImagePullBackOff   0             14m

### What did you expect to happen?

all pods a in Running Status when the update is done.

### How can we reproduce it (as minimally and precisely as possible)?

starting from K8s 1.25, apply the upgrade (rolling mode) to 1.26.
all the images needed by our pods are stored in a docker private registry and the secret to connect to it is in the namespace (default ns for the problematic pods)

### Kubernetes version

<details>

Client Version: version.Info{Major:"1", Minor:"26", GitVersion:"v1.26.9", GitCommit:"d1483fdf7a0578c83523bc1e2212a606a44fd71d", GitTreeState:"clean", BuildDate:"2023-10-07T07:04:59Z", GoVersion:"go1.21.2", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"26", GitVersion:"v1.26.9", GitCommit:"d1483fdf7a0578c83523bc1e2212a606a44fd71d", GitTreeState:"clean", BuildDate:"2023-09-13T11:25:26Z", GoVersion:"go1.20.8", Compiler:"gc", Platform:"linux/amd64"}

</details>

### Cloud provider

<details>

</details>

### OS version

<details>

NAME="Red Hat Enterprise Linux"
VERSION="8.8 (Ootpa)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="8.8"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Red Hat Enterprise Linux 8.8 (Ootpa)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:8::baseos"
HOME_URL="https://www.redhat.com/"
DOCUMENTATION_URL="https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 8"
REDHAT_BUGZILLA_PRODUCT_VERSION=8.8
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="8.8"

$ uname -a
Linux vmd-intk8sm2-01 4.18.0-477.21.1.el8_8.x86_64 

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>
Version:  0.1.0
RuntimeName:  cri-o
RuntimeVersion:  1.25.3
RuntimeApiVersion:  v1

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>
CNI = CALICO  v3.25.0
knative v1.10.0

</details>

**Comment by lallema** [comment]
sig/node

**Comment by neolit123** [comment]
i don't think this is a core k8s problem.

what could be deleting your secret in the default NS?
what tools are you using to setup the cluster and perform upgrade?

/kind support
/sig cluster-lifecycle

**Comment by lallema** [comment]
we don t use any specific tool.
we use kubeadm to create/setup the cluster (master / cp & worker nodes) and the rolling upgrade procedure to upgrade the cluster

**Comment by neolit123** [comment]
i don't know what you mean by rolling upgrade procedure, but if you are calling kubeamd upgrade:
https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade/
it does delete any user defined secrets in default NS.

**Comment by neolit123** [comment]
is this rolling upgrade procedure performed by a custom script in bash for example?

**Comment by lallema** [comment]
indeed, the secret is still in the ns (other pods are correctly restarted) but the knative-operator

default                knative-operator-5c746b59cd-lcvzq                                 0/1     ImagePullBackOff   0               67m
default                knative-operator-6649bc5fc7-cb62p                                 0/1     ImagePullBackOff   0               74m

(strange that i have 2 pods - appeared when i tried a kubecl rollout )

but the used one has:
 Warning  Failed     31s (x3 over 74s)  kubelet            Failed to pull image "intreg-dev1.lux.eib.org/knative-releases/knative.dev/operator/cmd/operator:v1.10.0": rpc error: code = Unknown desc = reading manifest v1.10.0 in intreg-dev1.lux.eib.org/knative-releases/knative.dev/operator/cmd/operator: unauthorized: authentication required
  Warning  Failed     31s (x3 over 74s)  kubelet            Error: ErrImagePull
  Normal   BackOff    8s (x5 over 74s)   kubelet            Back-off pulling image "intreg-dev1.lux.eib.org/knative-releases/knative.dev/operator/cmd/operator:v1.10.0"
  Warning  Failed     8s (x5 over 74s)   kubelet            Error: ImagePullBackOff

and that after the upgrade procedure (that went fine) 

Indeed, we are using this upgrade procedure you sent.

so, started to create from scratch a cluster in 1.25 (few weeks azgo), then applying our upg procedure with azure dev ops pipelines executing the commands in the doc

**Comment by lallema** [comment]
more details: CNI = CALICO v3.25.0 knative v1.10.0

**Comment by neolit123** [comment]
i don't see a bug in core kubernetes or in kubeadm here. it seems like k-native could be doing something.

you could try chatting with the knative community or the k8s community about this.
in general, for k8s we do not provide support on github:
https://github.com/kubernetes/kubernetes/blob/master/SUPPORT.md

thanks
/close

**Comment by lallema** [comment]
thanks ! i guess  we lost the imagePullSecret in the operator.yaml file :(
so secret exists (indeed) but not used by the operator
thanks anyway for your support !
